src/mcts/search.cc

```diff
@@ -1291,9 +1306,28 @@ void SearchWorker::ExtendNode(Node* node) {
       return;
     }
 
-    if (history_.Last().GetRepetitions() >= 2) {
+    // Mark two-fold repetitions as draws according to settings
+    const auto repetitions = history_.Last().GetRepetitions();
+    const auto twofolddrawlevel = params_.GetTwoFoldDrawLevel();
+    if (repetitions >= 2) {
       node->MakeTerminal(GameResult::DRAW);
       return;
+    } else if (repetitions == 1 && twofolddrawlevel > 0) {
+      if (twofolddrawlevel == 3) {
+        // always mark as draw
+        LOGFILE << "== marked level 3 twofold draw == depth: " << depth;
+        node->MakeTerminal(GameResult::DRAW, 0.0f, Node::Terminal::TwoFold);
```

Comment:
Nod. There seems to be some questionable changes here that were working fine in #1197, e.g., https://github.com/LeelaChessZero/lc0/pull/1197/files#diff-83b7141f4fa67bcfdf95188467895cb5R1224

Reply:
Thanks for catching that -- this probably is the reason for the PV being one ply too long. Fixed that.

Reply:
Thanks again, it now behaves as expected, and the correct PV is given.